pkg/openscap/openscap.go

```diff
@@ -0,0 +1,143 @@
+package openscap
+
+import (
+	"fmt"
+	docker "github.com/fsouza/go-dockerclient"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"syscall"
+)
+
+const (
+	CPE                = "oval:org.open-scap.cpe.rhel:def:"
+	CPE_DICT           = "/usr/share/openscap/cpe/openscap-cpe-oval.xml"
+	CVE_URL            = "https://www.redhat.com/security/data/metrics/ds/"
+	DIST_CVE_NAME_FMT  = "com.redhat.rhsa-RHEL%d.ds.xml.bz2"
+	ARF_RESULT_FILE    = "results-arf.xml"
+	TMP_DIR            = "/tmp"
+	LINUX              = "Linux"
+	IMAGE_SHORT_ID_LEN = 11
+)
+
+var (
+	RHEL_DIST_NUMBERS = [...]int{5, 6, 7}
+)
+
+// Scanner is the interface of OpenSCAP scanner
+type Scanner interface {
+	// Scan will scan
+	Scan() error
+}
+
+type defaultScanner struct {
+	// ChrootPath is the path where the image to be scanned is mounted
+	ChrootPath string
+	// CVEDir is the directory where the CVE file is saved
+	CVEDir string
+	// ArfResultFileName is the name of the arf report file
+	ArfResultFileName string
+	// Image is the metadata of the inspected image
+	Image *docker.Image
```

Comment:
Should `Image` be an attribute of the scanner or input to the scanner?  Seems like a  scanner could be used over and over if we made the image an input and the report an output.